Transmission 4.0.x does not build with fmt 10

For OpenBSD we recently updated our ports copy of fmt from 8.1.1 to 10.2.1. Transmission fails to build with fmt 10.

I noticed the main branch has updated to 10 already.

```
FAILED: libtransmission/CMakeFiles/transmission.dir/variant-json.cc.o
/home/ports/pobj/transmission-4.0.5/bin/c++ -DFMT_EXCEPTIONS=0 -DFMT_HEADER_ONLY=1 -DHAVE_FLOCK -DHAVE_MKDTEMP -DHAVE_PREAD -DHAVE_PWRITE -DHAVE_SO_REUSEPORT=1 -DHAVE_STATVFS -DHAVE_SYS_STATVFS_H -DPACKAGE_DATA_DIR=\"/usr/local/share\" -DPOSIX -DSYSTEM_MINIUPNP -DWIDE_INTEGER_DISABLE_FLOAT_INTEROP -DWIDE_INTEGER_DISABLE_IOSTREAM -DWITH_KQUEUE -DWITH_UTP -D__TRANSMISSION__ -I/home/ports/pobj/transmission-4.0.5/build-amd64/libtransmission -I/home/ports/pobj/transmission-4.0.5/transmission-4.0.5/third-party/libutp/include -I/home/ports/pobj/transmission-4.0.5/transmission-4.0.5/third-party/libb64/include -I/home/ports/pobj/transmission-4.0.5/transmission-4.0.5/third-party/jsonsl -I/home/ports/pobj/transmission-4.0.5/transmission-4.0.5/third-party/wildmat -isystem /usr/local/include -isystem /home/ports/pobj/transmission-4.0.5/transmission-4.0.5/cmake/../third-party/fast_float/include -isystem /home/ports/pobj/transmission-4.0.5/build-amd64/third-party/dht.bld/pfx/include -isystem /home/ports/pobj/transmission-4.0.5/transmission-4.0.5/cmake/../third-party/utfcpp/source -isystem /home/ports/pobj/transmission-4.0.5/transmission-4.0.5/cmake/../third-party/wide-integer -isystem /home/ports/pobj/transmission-4.0.5/transmission-4.0.5/cmake/../third-party/fmt/include -isystem /home/ports/pobj/transmission-4.0.5/build-amd64/third-party/libevent.bld/pfx/include -O2 -pipe -DNDEBUG -std=gnu++17 -W -Wall -Wextra -Wcast-align -Wexit-time-destructors -Wextra-semi -Wextra-semi-stmt -Wextra-tokens -Wfloat-equal -Wgnu -Winit-self -Wint-in-bool-context -Wmissing-format-attribute -Wnull-dereference -Wpointer-arith -Wredundant-decls -Wredundant-move -Wreorder-ctor -Wreturn-std-move -Wself-assign -Wself-move -Wsemicolon-before-method-body -Wsentinel -Wshadow -Wsign-compare -Wsometimes-uninitialized -Wstring-conversion -Wsuggest-destructor-override -Wsuggest-override -Wuninitialized -Wunreachable-code -Wunused -Wunused-const-variable -Wunused-parameter -Wunused-result -Wwrite-strings -Wformat-security -MD -MT libtransmission/CMakeFiles/transmission.dir/variant-json.cc.o -MF libtransmission/CMakeFiles/transmission.dir/variant-json.cc.o.d -o libtransmission/CMakeFiles/transmission.dir/variant-json.cc.o -c /home/ports/pobj/transmission-4.0.5/transmission-4.0.5/libtransmission/variant-json.cc
In file included from /home/ports/pobj/transmission-4.0.5/transmission-4.0.5/libtransmission/variant-json.cc:20:
In file included from /usr/local/include/fmt/compile.h:11:
In file included from /usr/local/include/fmt/format.h:49:
/usr/local/include/fmt/core.h:1600:63: error: implicit instantiation of undefined template 'fmt::detail::type_is_unformattable_for<fmt::detail::named_arg<char, jsonsl_error_t>, char>'
    type_is_unformattable_for<T, typename Context::char_type> _;
                                                              ^
/usr/local/include/fmt/core.h:1842:23: note: in instantiation of function template specialization 'fmt::detail::make_arg<true, fmt::basic_format_context<fmt::appender, char>, fmt::detail::named_arg<char, jsonsl_error_t>, 0>' requested here
        data_{detail::make_arg<is_packed, Context>(args)...} {
                      ^
/usr/local/include/fmt/core.h:1860:10: note: in instantiation of function template specialization 'fmt::format_arg_store<fmt::basic_format_context<fmt::appender, char>, fmt::detail::named_arg<char, unsigned long>, fmt::detail::named_arg<char, std::string_view>, fmt::detail::named_arg<char, const char *>, fmt::detail::named_arg<char, jsonsl_error_t>>::format_arg_store<fmt::detail::named_arg<char, unsigned long>, fmt::detail::named_arg<char, std::string_view>, fmt::detail::named_arg<char, const char *>, fmt::detail::named_arg<char, jsonsl_error_t>>' requested here
  return {args...};
         ^
/usr/local/include/fmt/core.h:2835:28: note: in instantiation of function template specialization 'fmt::make_format_args<fmt::basic_format_context<fmt::appender, char>, fmt::detail::named_arg<char, unsigned long>, fmt::detail::named_arg<char, std::string_view>, fmt::detail::named_arg<char, const char *>, fmt::detail::named_arg<char, jsonsl_error_t>>' requested here
  return vformat(fmt, fmt::make_format_args(args...));
                           ^
/home/ports/pobj/transmission-4.0.5/transmission-4.0.5/libtransmission/variant-json.cc:97:14: note: in instantiation of function template specialization 'fmt::format<fmt::detail::named_arg<char, unsigned long>, fmt::detail::named_arg<char, std::string_view>, fmt::detail::named_arg<char, const char *>, fmt::detail::named_arg<char, jsonsl_error_t>>' requested here
        fmt::format(
             ^
/usr/local/include/fmt/core.h:1578:45: note: template is declared here
template <typename T, typename Char> struct type_is_unformattable_for;
                                            ^
/usr/local/include/fmt/core.h:1603:3: error: static assertion failed due to requirement 'formattable': Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
  static_assert(
  ^
2 errors generated.
ninja: build stopped: subcommand failed.
```

--- a/gtk/Application.cc
+++ b/gtk/Application.cc
@@ -395,7 +395,7 @@ void register_magnet_link_handler()
             _("Couldn't register Transmission as a {content_type} handler: {error} ({error_code})"),
             fmt::arg("content_type", content_type),
             fmt::arg("error", e.what()),
-            fmt::arg("error_code", e.code())));
+            fmt::arg("error_code", static_cast<int>(e.code()))));
     }
 }
 

--- a/libtransmission/file-posix.cc
+++ b/libtransmission/file-posix.cc
@@ -560,7 +560,7 @@ tr_sys_file_t tr_sys_file_get_std(tr_std_sys_file_t std_file, tr_error** error)
         break;
 
     default:
-        TR_ASSERT_MSG(false, fmt::format(FMT_STRING("unknown standard file {:d}"), std_file));
+        TR_ASSERT_MSG(false, fmt::format(FMT_STRING("unknown standard file {:d}"), static_cast<int>(std_file)));
         tr_error_set_from_errno(error, EINVAL);
     }
 

--- a/libtransmission/variant-json.cc
+++ b/libtransmission/variant-json.cc
@@ -99,7 +99,7 @@ void error_handler(jsonsl_t jsn, jsonsl_error_t error, jsonsl_state_st* /*state*
             fmt::arg("position", jsn->pos),
             fmt::arg("text", std::string_view{ buf, std::min(size_t{ 16U }, data->size - jsn->pos) }),
             fmt::arg("error", jsonsl_strerror(error)),
-            fmt::arg("error_code", error)));
+            fmt::arg("error_code", static_cast<int>(error))));
 }
 
 int error_callback(jsonsl_t jsn, jsonsl_error_t error, struct jsonsl_state_st* state, jsonsl_char_t* at)

